File: lab2/source/cr.cpp
#include "operations.h"
#include "factory.h"

class Cr: public Operations{
public:
    void Operation(std::stack<int> &stack, std::istream& fin) override {
        Empty_check(stack);
        int previous_top = stack.top();
        stack.pop();
        Empty_check(stack);
        std::cout<< previous_top << "\n" << stack.top() << std::endl;
        stack.push(previous_top);
    }
};

namespace {
    Operations* createCr() {
        return new Cr;
    }
    bool x = OperationsFactory::getInstance()->RegisterOperation(std::string( "cr"), createCr);
}

File: jaba/lab1/src/main/java/org/example/Cr.java
package org.example;

import java.io.OutputStream;
import java.util.Scanner;
import java.util.Stack;

public class Cr extends Operations{
    @Override
    public void Operation(Stack<Integer> stack, Scanner scanner, OutputStream s){
        emptyCheck(stack);
        int previous_top = stack.pop();
        emptyCheck(stack);
        System.out.printf("%d%n%d%n", previous_top, stack.peek());
        stack.push(previous_top);
    }
}
